components/private/ShapeLibrary/src/MDCCutCornerTreatment.h

```diff
@@ -0,0 +1,41 @@
+/*
+ Copyright 2018-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <CoreGraphics/CoreGraphics.h>
+
+#import "MaterialShapes.h"
+
+@interface MDCCutCornerTreatment : MDCCornerTreatment
+
+/**
+ The cut of the corner.
+ */
+@property(nonatomic, assign) CGFloat cut;
+
+/**
+ Initializes an MDCCutCornerTreatment instance with a given cut.
+ */
+- (nonnull instancetype)initWithCut:(CGFloat)cut NS_DESIGNATED_INITIALIZER;
+
+/**
+ Initializes an MDCCutCornerTreatment instance with a cut of zero.
+ */
+- (nonnull instancetype)init;
+
+- (nullable instancetype)initWithCoder:(nonnull NSCoder *)aDecoder NS_DESIGNATED_INITIALIZER;
+
+
+@end
```

Comment:
Small nit: this looks like good code that could have been broken out in to a separate, smaller PR.

Reply:
Don't worry about it for this PR though unless this PR for some reason ends up stuck in review.